python/lsst/obs/subaru/objectMasks.py

```diff
@@ -0,0 +1,132 @@
+        """
+
+        log = pexLog.getDefaultLog().createChildLog("ObjectMaskCatalog")
+
+        brightObjects = ObjectMaskCatalog()
+        checkedWcsIsFk5 = False
+
+        with open(fileName) as fd:
+            for lineNo, line in enumerate(fd.readlines(), 1):
+                line = line.rstrip()
+
+                if re.search(r"^\s*#", line):
+                    #
+                    # Parse any line of the form "# key : value" and put them into the metadata.
+                    #
+                    # As noted in HSC-1342, we expect to see at least the keys
+                    #   tract patch filter
+                    # (they will be converted to lower case, in case we switch to using FITS,
+                    # whose keys are case insensitive).  The value of these three keys will be checked,
+                    # so get them right!
+                    #
+                    mat = re.search(r"^\s*#\s*([a-zA-Z][a-zA-Z0-9_]+)\s*:\s*(.*)", line)
+                    if mat:
+                        key, value = mat.group(1).lower(), mat.group(2)
+                        if key == "tract":
+                            value = int(value)
+
+                        brightObjects.table.getMetadata().set(key, value)
+
+                line = re.sub(r"^\s*#.*", "", line)
+                if not line:
+                    continue
+
+                if re.search(r"^\s*wcs\s*;\s*fk5\s*$", line, re.IGNORECASE):
+                    checkedWcsIsFk5 = True
+                    continue
+
+                mat = re.search(r"^\s*circle(?:\s+|\s*\(\s*)"
+                                "(\d+(?:\.\d*)([d]*))" "(?:\s+|\s*,\s*)"
+                                "([+-]?\d+(?:\.\d*)([d]*))" "(?:\s+|\s*,\s*)"
+                                "(\d+(?:\.\d*))([d'\"]*)" "(?:\s*|\s*\)\s*)"
+                                "\s*#\s*ID:\s*(\d+)" "\s*$"
+                                , line)
+                if mat:
+                    ra, raUnit, dec, decUnit, radius, radiusUnit, _id = mat.groups()
+
+                    _id = int(_id)
+                    ra =     convertToAngle(ra,     raUnit,     "ra",     fileName, lineNo)
+                    dec =    convertToAngle(dec,    decUnit,    "dec",    fileName, lineNo)
+                    radius = convertToAngle(radius, radiusUnit, "radius", fileName, lineNo)
+
+                    rec = brightObjects.addNew()
+                    # N.b. rec["coord"] = Coord is not supported, so we have to use the setter
+                    rec["id"] = _id
+                    rec.setCoord(afwCoord.Fk5Coord(ra, dec))
+                    rec["radius"] = radius
+                else:
+                    log.warn("Unexpected line \"%s\" at %s:%d" % (line, fileName, lineNo))
+
+        if not checkedWcsIsFk5:
+            raise RuntimeError("Expected to see a line specifying an fk5 wcs")
+
+        brightObjects._catalog = brightObjects._catalog.copy(True) # allow numpy access to columns
```

Comment:
Strictly, the deep copy makes it contiguous in memory so that you can use a {{ColumnView}} which can be exposed to NumPy. I think an extra couple of words here would help make that clear.